### ISSUE TITLE
Limited number of groups in the menu?

### ISSUE THREAD

**Issue by mforkel** [Morikko/sync-tab-groups#117]
Could it be that there is a limit to the number of groups shown in the pull-down menu (and in the group manager)? When I open a new Firefox 60.0 window, create some tabs, and click on 'Group Invisible' of Sync Tab Groups 0.6.7, a new entry for an unnamed groups shows up in the menu, but only for a split second. The prefix '[Unnamed Group 33]' to the browser windows's title does not disappear again, though. Magic number 32?

If there is a limit, how can I raise it?

**Comment by Morikko** [comment]
There is no limit, with next STG version (v0.6.9) could you tell me if an error occurs, a notification will show if it is the case.

**Comment by mforkel** [comment]
Looking into the browser console, the  click on 'Group Invisible' results in
```
11:09:04.404 GroupManager.checkCorruptedGroups has detected a corrupted groups:  1 groupmanager.js:1012:5
	GroupManager.checkCorruptedGroups moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/core/groupmanager.js:1012:5
	GroupManager.store moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/core/groupmanager.js:920:8
	GroupManager.initEventListener/</< moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/core/groupmanager.js:934:7
	TaskManager.RepeatedTask.prototype.add moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/utils/repeatedtaskmanager.js:48:13
	GroupManager.initEventListener/< moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/core/groupmanager.js:933:5
	EventListener.prototype.fire/< moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/utils/eventlistener.js:13:7
	EventListener.prototype.fire moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/utils/eventlistener.js:12:5
	GroupManager.initEventListener/< moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/core/groupmanager.js:941:5
	EventListener.prototype.fire/< moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/utils/eventlistener.js:13:7
	EventListener.prototype.fire moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/utils/eventlistener.js:12:5
	GroupManager.addGroupWithTab moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/core/groupmanager.js:743:3
	WindowManager.addGroupFromWindow moz-extension://5c532073-f272-4df1-9e84-e538674b7672/background/core/windowmanager.js:538:22
11:09:04.408 Tried to correct corruption... 1 groupmanager.js:1016:7
11:09:06.946 Content Security Policy: Die Einstellungen der Seite haben das Laden einer Ressource auf self blockiert ("script-src"). Source: call to eval() or related function blocked by CSP. 1 contextmenus.js:82
```
The last entry propably translates to ``Content Security Policy: The page's settings blocked the loading of a resource at self``

I can open all groups that were closed, but I see console entries like
```
11:27:18.687 Error: ID already exists: stg-move-tab-group-8 1 ext-menus.js
```
and
```
11:27:49.516 Error: Invalid tab ID: -1 1 ext-browser.js
```

**Comment by mforkel** [comment]
I managed to solve my problem by:
- exporting all groups
- closing all browser windows with open groups
- moving all tabs of the 2 unnamed groups 33 I had into new browser windows
- removing all groups
- importing the (old) groups

I can now create new groups again.

**Comment by Morikko** [comment]
Version v0.6.9 just landed: I fix #116 and also improve the corruption checker. Before, it triggered sometimes on wrong positives. As a consequence, STG was loaded the group from the disk and removing the changes (the ones added with "Group Visible").

Normally, it is fixed, let me know if things are going well the next days.
